### PR TITLE
Update the hardcoded dotnet-version with channels

### PR DESCRIPTION
**Pull request by kunalspathak**

### CODE REVIEW

**Comment by kunalspathak** [comment]
Scheduled an internal superpmi-collect run to see if it fixes the failing benchmarks collection.

**Comment by kunalspathak** [comment]
@BruceForstall - this seems to fix the problem.

File: src/coreclr/scripts/superpmi_collect_setup.py [hunk 1]
```diff
@@ -405,7 +405,7 @@ def setup_microbenchmark(workitem_directory, arch):
         # have not published yet. As a result, we hit errors of "dotnet restore". As a workaround, hard code the
         # working version until we move to ".NET 8" in the script.
         run_command(
-            get_python_name() + [dotnet_install_script, "install", "--dotnet-versions", "8.0.100-alpha.1.22558.2", "--architecture", arch, "--install-dir",
+            get_python_name() + [dotnet_install_script, "install", "-channels", "8.0-preview", "--architecture", arch, "--install-dir",
```
Comment (BruceForstall) [inline src/coreclr/scripts/superpmi_collect_setup.py:408]: Shouldn't this have two dashes?
```suggestion
            get_python_name() + [dotnet_install_script, "install", "--channels", "8.0-preview", "--architecture", arch, "--install-dir",
```
?

Comment (kunalspathak) [reply src/coreclr/scripts/superpmi_collect_setup.py:408]: Ah yes, I pushed the fixed commit to the internal site that led to successful spmi collection, but forgot to push it here.

**Review by BruceForstall — APPROVED** [review]
Do you know what the difference is between `--channels release/8.0` that you used and `--channels 8.0-preview` that I used (that Scott originally suggested)? It looks like yours installed `8.0.100-preview.4.23174.9`.

**Comment by kunalspathak** [comment]
When I tried locally, `8.0-preview` was not a valid parameter. I got this error and hence I changed it to `release/8.0`.

```
e:\git\performance\scripts>python dotnet.py install --channels 8.0-preview --install-dir f:\temp\todelete
usage: dotnet.py install [-h]
                         [--channels {main,master,8.0,release/8.0,nativeaot8.0,7.0,release/7.0-rc2,release/7.0-rc1,nativeaot7.0,release/7.0,6.0,release/6.0,nativeaot6.0,release/3.1.3xx,release/3.1.2xx,release/3.1.1xx,3.1,LTS,net48} [{main,master,8.0,release/8.0,nativeaot8.0,7.0,release/7.0-rc2,release/7.0-rc1,nativeaot7.0,release/7.0,6.0,release/6.0,nativeaot6.0,release/3.1.3xx,release/3.1.2xx,release/3.1.1xx,3.1,LTS,net48} ...]]
                         [--architecture {x64,x86,arm,arm64}] [--dotnet-versions DOTNET_VERSIONS [DOTNET_VERSIONS ...]] [--install-dir INSTALL_DIR] [-v]
dotnet.py install: error: argument --channels: invalid choice: '8.0-preview' (choose from 'main', 'master', '8.0', 'release/8.0', 'nativeaot8.0', '7.0', 'release/7.0-rc2', 'release/7.0-rc1', 'nativeaot7.0', 'release/7.0', '6.0', 'release/6.0', 'nativeaot6.0', 'release/3.1.3xx', 'release/3.1.2xx', 'release/3.1.1xx', '3.1', 'LTS', 'net48')
```

**Comment by kunalspathak** [comment]
> Scott added that channel on 3/24:

Got it. https://github.com/dotnet/runtime/pull/83984

**Pull request merged**